packages/utils/src/findSchemaDefinition.ts

```diff
@@ -18,37 +18,69 @@ export function splitKeyElementFromObject(key: string, object: GenericObjectType
   return [remaining, value];
 }
 
-/** Given the name of a `$ref` from within a schema, using the `rootSchema`, look up and return the sub-schema using the
- * path provided by that reference. If `#` is not the first character of the reference, or the path does not exist in
- * the schema, then throw an Error. Otherwise return the sub-schema. Also deals with nested `$ref`s in the sub-schema.
+/** Given the name of a `$ref` from within a schema, using the `rootSchema`, recursively look up and return the
+ * sub-schema using the path provided by that reference. If `#` is not the first character of the reference, the path
+ * does not exist in the schema, or the reference resolves circularly back to itself, then throw an Error.
+ * Otherwise return the sub-schema. Also deals with nested `$ref`s in the sub-schema.
  *
  * @param $ref - The ref string for which the schema definition is desired
  * @param [rootSchema={}] - The root schema in which to search for the definition
+ * @param recurseList - List of $refs already resolved to prevent recursion
  * @returns - The sub-schema within the `rootSchema` which matches the `$ref` if it exists
- * @throws - Error indicating that no schema for that reference exists
+ * @throws - Error indicating that no schema for that reference could be resolved
  */
-export default function findSchemaDefinition<S extends StrictRJSFSchema = RJSFSchema>(
+export function findSchemaDefinitionRecursive<S extends StrictRJSFSchema = RJSFSchema>(
   $ref?: string,
```

Comment:
By defaulting to an empty string here you can simplify the code below a little bit
```suggestion
  $ref?: string = '',
```